components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/dao/IdPManagementDAO.java

```diff
@@ -1073,15 +1114,23 @@ private void updateFederatedAuthenticatorConfig(FederatedAuthenticatorConfig new
                 }
             }
 
+            String authnConfigName = newFederatedAuthenticatorConfig.getName();
+            ArrayList<String> federatedAuthenticatorConfidentialPropertyNames =
```

Comment:
```suggestion
            List<String> federatedAuthenticatorConfidentialPropertyNames =
```

Reply:
check in other places as well. when defining, we should use the base interface or class

Reply:
fixed with https://github.com/wso2/carbon-identity-framework/pull/3798/commits/6a2f91760f8d0931e1bc54964973f61ba589d79a